grype/presenter/cyclonedx/vuln-extension.go

```diff
@@ -52,10 +57,15 @@ type Vulnerability struct {
 	Advisories *Advisories `xml:"v:advisories"`
 }
 
+// Vulnerabilities contains all the vulnerabilities for a component
+type Vulnerabilities struct {
+	Vulnerability []Vulnerability `xml:"v:vulnerability,omitempty"`
+}
+
 // Component represents the a single software/package that has vulnerabilities.
 type Component struct {
 	syftCDX.Component
-	Vulnerabilities []Vulnerability `xml:"v:vulnerabilities"`
```

Comment:
This change allows passing the validation, basically it is nesting vulnerability into vulnerabilities

Reply:
This approach is still leaving empty tags in the generated output:
```xml
    <component type="library">
      <name>base-passwd</name>
      <version>3.5.47</version>
      <v:vulnerabilities></v:vulnerabilities>
    </component>
```
where more correct behavior would be to not include empty tags:
```xml
    <component type="library">
      <name>base-passwd</name>
      <version>3.5.47</version>
    </component>
```
... both approaches are within-spec relative to CycloneDX, but the latter is preferred.

The `omitempty` struct tag option for `xml` has the following behavior (from https://golang.org/pkg/encoding/xml/#Marshal):
```
a field with a tag including the "omitempty" option is omitted
  if the field value is empty. The empty values are false, 0, any
  nil pointer or interface value, and any array, slice, map, or
  string of length zero.
``` 
which means that empty structs are treated as non-empty values (so `omitempty` will not omit empty `Vulnerability` instances). If instead the type was `[]Vulnerability` or `*Vulnerabilities` I believe the xml tags would be omitted.

This is the case for at least `rating` and `severity` tags (maybe more)

Reply:
Ahhh interesting, I thought I was catching them but I guess it wasn't sufficient. Will address this